examples/cs.py

```diff
@@ -193,4 +209,74 @@ def dcs(w):
     ionization_osi = ionization_grp.create_dataset("outer_shells", data=s.elf_file.get_outer_shells().to('eV'))
     ionization_osi.attrs['units'] = 'eV'
 
+    # electron range
+    e_ran = np.logspace(-2, 3, 129) * units.eV
+    p_ran = np.linspace(0.0, 1.0, 1024)
+
+    electron_range_grp = outfile.create_group("electron_range")
+    ran_energies = electron_range_grp.create_dataset("energy", data=e_ran.to('eV'))
+    ran_energies.attrs['units'] = 'eV'
+    ran_range = electron_range_grp.create_dataset("range", e_ran.shape)
+    ran_range.attrs['units'] = 'm'
+
+    ran_tmfp = np.zeros(e_ran.shape) * units.m;
+    tmfpmax = 0. * units.m
+    print("# Computing transport mean free path.")
+    for i, K in enumerate(e_ran):
+        def dcs(theta):
+            return elastic_cs_fn(theta, K) * (1 - np.cos(theta)) * s.rho_n
+        inv_tmfp = compute_elastic_tcs(dcs, p_ran)
+        tmfp = 1./inv_tmfp
+        if ( K >= s.band_structure.barrier):
+            # make sure the tmfp is monotonously increasing for energies above
+            # the vacuum potential barrier
+            if (tmfp < tmfpmax):
+                tmfp = tmfpmax
+            else:
+                tmfpmax = tmfp
+        else:
+            tmfpmax = tmfp
+        ran_tmfp[i] = tmfp
+        print('.', end='', flush=True)
+    print()
+
+    rangemax = 0. * units.m
+    tmprange = np.zeros(e_ran.shape) * units.m
+    print("# Computing inelastic electron range.")
+    for i, K in enumerate(e_ran):
+        ran_energies[i] = K.to('eV')
+        if (K < s.band_structure.barrier):
+            ran_range[i] = 0. * units.m
+        else:
+            w0_max = K-s.band_structure.fermi # it is not possible to lose so
+            # much energy that the primary electron ends up below the Fermi
+            # level in an inelastic scattering event
+
+            def dcs(w):
+                return inelastic_cs_fn(s)(K, w) * s.rho_n
+            tcs = compute_inelastic_tcs(dcs, p_inel,
+                s.elf_file.get_min_energy(), w0_max,
+                s.elf_file.get_min_energy_interval())
+            tmprange[i] = 1/tcs
+            for j, omega in enumerate(e_ran):
+                if (omega < K/2 and omega < K-s.band_structure.fermi):
+                    if j==0:
```

Comment:
This may be more of a personal opinion, but I would prefer `omega_old = 0 if j==0 else e_ran[j-1]` here. These ternary operators tend to lead to excessively long lines, so one should be careful with them, but I think this is a case where it helps the code clarity.